File: phpPOO/POOAula11/Professor.php
<?php
require_once './Pessoa.php';
class Professor extends Pessoa{
    private $especial;
    private $salario;
     public final function receberAumento($a){
         $this->setSalario($this->getSalario()+ $a);
     }
     
     public final function mostrarSalario(){
         echo "<p>Salario atual de ".$this->getNome()." ".$this->getSalario()."</p>";
     }
     public function getEspecial() {
         return $this->especial;
     }

     public function getSalario() {
         return $this->salario;
     }

     public function setEspecial($especial): void {
         $this->especial = $especial;
     }

     public function setSalario($salario): void {
         $this->salario = $salario;
     }


}

File: javaPOO/POOAula11/src/pooaula11/Professor.java
package pooaula11;
public class Professor extends Aluno{
    private String especialidade;
    private float salario;

    public String getEspecialidade() {
        return especialidade;
    }

    public void setEspecialidade(String especialidade) {
        this.especialidade = especialidade;
    }

    public float getSalario() {
        return salario;
    }

    public void setSalario(float salario) {
        this.salario = salario;
    }
    
    public float mostrarSalario(){
        return this.salario;
    }
    
    public float receberAumento(float a){
        this.setSalario(this.getSalario()+a);
        return this.salario;
    }    
}
